lib/src/extensions.dart

```diff
@@ -427,6 +427,14 @@ extension NullableAstNodeExtension on AstNode? {
   }
 }
 
+extension TokenExtension on Token? {
+  bool get isFinal {
+    var self = this;
+    if (self == null) return false;
+    return self.keyword == Keyword.FINAL;
```

Comment:
Not that it matters, but I think you could just write `return this?.keyword == Keyword.FINAL;` (and drop the two lines above).

Reply:
Ah, but even better still...

```dart
extension TokenExtension on Token? {
  bool get isFinal => this?.keyword == Keyword.FINAL;
}
```

Thanks!